internal/pkg/_presentation/api/ngsi-ld/context/context.go

```diff
@@ -0,0 +1,124 @@
+package context
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/diwise/api-smartwater/internal/pkg/application"
+	"github.com/diwise/api-smartwater/internal/pkg/infrastructure/repositories/models"
+	"github.com/diwise/ngsi-ld-golang/pkg/datamodels/fiware"
+	"github.com/diwise/ngsi-ld-golang/pkg/ngsi-ld"
+	"github.com/rs/zerolog"
+)
+
+type contextSource struct {
+	app application.Application
+	log zerolog.Logger
+}
+
+//CreateSource instantiates and returns a Fiware ContextSource that wraps the provided application interface
+func CreateSource(app application.Application, log zerolog.Logger) ngsi.ContextSource {
+	return &contextSource{
+		app: app,
+		log: log,
+	}
+}
+
+func (cs contextSource) CreateEntity(typeName, entityID string, req ngsi.Request) error {
+	if typeName != fiware.WaterConsumptionObservedTypeName {
+		errorMessage := fmt.Sprintf("entity type %s not supported", typeName)
+		cs.log.Error().Msg(errorMessage)
+		return errors.New(errorMessage)
+	}
+
+	wco := &fiware.WaterConsumptionObserved{}
+	err := req.DecodeBodyInto(wco)
+	if err != nil {
+		return err
+	}
+
+	observedAt, err := time.Parse(time.RFC3339, wco.WaterConsumption.ObservedAt)
+	if err != nil {
+		return err
+	}
+
+	err = cs.app.UpdateWaterConsumption(entityID, wco.WaterConsumption.Value, observedAt)
```

Comment:
You could return immediately here. No point in saving the return value in a variable and then return that on the next line.